### PR TITLE
BAVL-419 refactoring integration tests, no functional changes to API.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AppointmentsService.kt [hunk 1]
```diff
@@ -25,6 +25,7 @@ class AppointmentsService(
   fun createAppointmentsForCourt(videoBooking: VideoBooking, prisoner: PrisonerDetails, user: User) {
     checkCourtAppointments(prisoner.appointments, prisoner.prisonCode!!, user)
 
+    // We don't need to do a check for non-prison users as this is covered by the overlapping appointment check.
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AppointmentsService.kt:28]: Turns out this is covered for other users in the overlapping appointment checking (which is not done for prison users), added comment to further explain why it is the way it is.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/VideoLinkBookingIntegrationTest.kt [hunk 2]
```diff
@@ -114,14 +115,21 @@ class VideoLinkBookingIntegrationTest : SqsIntegrationTestBase() {
   @Autowired
   private lateinit var bookingHistoryRepository: BookingHistoryRepository
 
+  @BeforeEach
+  fun setUpLocations() {
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/VideoLinkBookingIntegrationTest.kt:119]: Pulled up to here to remove duplication.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/MigrationController.kt [hunk 3]
```diff
@@ -14,7 +14,7 @@ import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.handle
 
 @Tag(name = "Migrate a video link booking from old BVLS")
 @RestController
-@RequestMapping(value = ["migrate"], produces = [MediaType.APPLICATION_JSON_VALUE])
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/MigrationController.kt:17]: Just noticed this is not needed for this temporary migration controller.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/IntegrationTestBase.kt [hunk 4]
```diff
@@ -60,6 +65,15 @@ abstract class IntegrationTestBase {
     stubUser(PROBATION_USER)
   }
 
+  @BeforeEach
+  fun `stub default locations`() {
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/IntegrationTestBase.kt:69]: Stubbing some default locations here to reduce the duplication across tests.

We could look at doing this for other data e.g. some default prisoners etc.

This should help keep the integration tests a bit cleaner and lighter 🤞

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**